spec/consensus/proposer-based-timestamp/README.md

```diff
@@ -1,20 +1,158 @@
-# Proposer-Based Timestamps
+# Proposer-Based Timestamps (PBTS)
 
-This section describes a version of the Tendermint consensus protocol,
-which uses proposer-based timestamps.
+This section describes a version of the Tendermint consensus protocol
+that uses proposer-based timestamps.
 
-## Contents
+## Context
 
-- [Proposer-Based Time][main] (entry point)
-- [Part I - System Model and Properties][sysmodel]
-- [Part II - Protocol Specification][algorithm]
-- [TLA+ Specification][proposertla]
+Tendermint provides a deterministic, Byzantine fault-tolerant, source of time,
+defined by the `Time` field present in the headers of committed blocks,
+representing the block's timestamp.
 
+In the current consensus implementation, the timestamp of a block is
+computed by the [`BFTTime`][bfttime] algorithm:
 
-[algorithm]: ./pbts-algorithm_002_draft.md
+- Validators timestamps the `Precommit` messages they broadcast.
+Timestamps are retrieved from the validator's local clocks,
+with the only restriction that they must be **monotonic**:
+
+	- The timestamp of a `Precommit` message voting for a block
+	cannot be earlier than the `Time` field of that block;
+
+- The timestamp of a block is deterministically computed from the timestamps of
+a set of `Precommit` messages that certify the commit of the previous block.
+This certificate, a set of `Precommit` messages from a round of the previous height,
+is selected by the block's proposer and stored in the `Commit` field of the block:
+
+	- The block timestamp is the *median* of the timestamps of the `Precommit` messages
+	included in the `Commit` field, weighted by their voting power.
+	Block timestamps are **monotonic** because
+	timestamps of valid `Precommit` messages are monotonic;
+
+Assuming that the voting power controlled by Byzantine validators is bounded by `f`,
+the cumulative voting power of any valid `Commit` set must be at least `2f+1`.
+As a result, the timestamp computed by `BFTTime` is not influenced by Byzantine validators,
+as the weighted median of `Commit` timestamps comes from the clock of a non-faulty validator.
```

Comment:
Not sure I understand why this is true. If `f` of the voting power is controlled by byzantine validators, couldn't they shift the median by generating faulty timestamps?

Reply:
The point here is: if Byzantine validators produce timestamps in the past (or the future), when the `2f + 1` timestamps are ordered, the `f+1`th timestamp will not be in the past (or future).